### PR TITLE
fix(m365): fix opening outlook channel config page no response

### PR DESCRIPTION
**Pull request by kuojianlu**
To fix https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/14423884.

E2E TEST: N/A

### CODE REVIEW

File: packages/cli/src/cmds/preview/launch.ts [hunk 1]
```diff
@@ -99,24 +99,52 @@ export async function openUrlWithNewProfile(url: string): Promise<boolean> {
...
+      {
+        name: open.apps.edge,
+        arguments: [`--user-data-dir=${profileFolderPath}`],
+      },
+      {
+        name: open.apps.firefox,
+        arguments: ["-profile", profileFolderPath],
+      },
+    ];
+    for (const app of apps) {
```
Comment (dooriya) [inline packages/cli/src/cmds/preview/launch.ts:142]: Is that possible to select browser when preview, like the local debug?

**Review by a1exwang — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

File: packages/cli/src/cmds/preview/launch.ts [hunk 2]
```diff
@@ -99,24 +99,52 @@ export async function openUrlWithNewProfile(url: string): Promise<boolean> {
...
+            resolve(false);
+          }
+          resolve(true);
+        } catch {
+          resolve(false);
+        }
+      });
+    };
+
+    const apps = [
```
Comment (dooriya) [inline packages/cli/src/cmds/preview/launch.ts:128]: Can we move this to global const so CLI and extension can share?

File: packages/cli/src/cmds/preview/launch.ts [hunk 3]
```diff
@@ -99,24 +99,52 @@ export async function openUrlWithNewProfile(url: string): Promise<boolean> {
...
+        name: open.apps.firefox,
+        arguments: ["-profile", profileFolderPath],
+      },
+    ];
+    for (const app of apps) {
+      if (await tryToOpen(url, app)) {
+        return true;
+      }
+    }
+    return false;
```
Comment (dooriya) [inline packages/cli/src/cmds/preview/launch.ts:147]: Will we provide error message to user if none of the expected browser found?

File: packages/cli/src/cmds/preview/launch.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/cli/src/cmds/preview/launch.ts:128]: The UX for manually configuring Outlook channel is a workaround and will be removed once App Studio API is ready. So I just copy/paste this piece of code for simplicity.

File: packages/cli/src/cmds/preview/launch.ts (hunk 3 above)
Comment (kuojianlu) [reply packages/cli/src/cmds/preview/launch.ts:147]: If `openUrlWithNewProfile` fails, will directly open the url in the default browser, so no error message needed. Introducing `openUrlWithNewProfile` is to enforce users to always login in to bot framework portal.

File: packages/cli/src/cmds/preview/launch.ts (hunk 1 above)
Comment (kuojianlu) [reply packages/cli/src/cmds/preview/launch.ts:142]: The browser selected is only for sideloading Teams/Outlook.
For extension part, the browser type cannot be known outside debug provider. So we have to try each browser. And for CLI part, to keep consistent with the extension.

**Review by dooriya — APPROVED** [review]

File: packages/cli/src/cmds/preview/launch.ts (hunk 3 above)
Comment (dooriya) [reply packages/cli/src/cmds/preview/launch.ts:147]: Make sense. Thanks for the explanation~

**Pull request merged**